src/main/java/com/whiteclarkegroup/liquibaselinter/config/ConfigLoader.java

```diff
@@ -43,39 +44,50 @@ private Config loadConfig(ResourceAccessor resourceAccessor, String path) throws
         final Set<InputStream> resourcesAsStream = resourceAccessor.getResourcesAsStream(path);
         if (resourcesAsStream != null && !resourcesAsStream.isEmpty()) {
             try (InputStream inputStream = resourcesAsStream.iterator().next()) {
-                return loadImports(resourceAccessor, Config.fromInputStream(inputStream));
+                final Config config = Config.fromInputStream(inputStream);
+                return loadImports(resourceAccessor, config, config.getImports());
             }
         }
         return null;
     }
 
-    private Config loadImports(ResourceAccessor resourceAccessor, Config config) {
-        if (config.getImports() == null || config.getImports().isEmpty()) {
-            return config;
-        }
-        final ListMultimap<String, RuleConfig> configRules = Optional.ofNullable(config.getRules()).orElse(ArrayListMultimap.create());
-        final ImmutableListMultimap.Builder<String, RuleConfig> combinedRules = new ImmutableListMultimap.Builder<>();
-        combinedRules.putAll(configRules);
+    private Config loadImports(ResourceAccessor resourceAccessor, Config config, List<String> imports) {
+        final ListMultimap<String, RuleConfig> rules = ArrayListMultimap.create();
+        final ListMultimap<String, Reporter> reporting = ArrayListMultimap.create();
 
-        config.getImports().forEach(path -> {
-            try {
-                final Config importedConfig = loadConfig(resourceAccessor, path);
-                if (importedConfig == null) {
-                    throw new UnexpectedLiquibaseException("Failed to load imported lq lint config file: " + path);
-                }
-                final ListMultimap<String, RuleConfig> importedRules = importedConfig.getRules();
-                if (importedRules != null) {
-                    importedRules.asMap().forEach((key, rules) -> {
-                        if (!configRules.containsKey(key)) {
-                            combinedRules.putAll(key, rules);
-                        }
-                    });
+        if (config.getRules() != null) {
+            rules.putAll(config.getRules());
+        }
+        if (config.getReporting() != null) {
+            reporting.putAll(config.getReporting());
+        }
+        if (imports != null) {
+            for (String path : imports) {
+                try {
+                    final Config importedConfig = loadConfig(resourceAccessor, path);
+                    combine(config.getRules(), importedConfig.getRules(), rules);
```

Comment:
Possible `NullPointerException` here as `loadConfig` can return `null`

Reply:
That is why I catch `NullPointerException` in the catch block. It needs to result in an `UnexpectedLiquibaseException`, so is it worth checking for null and duplicating the throw below?